library/src/main/java/com/nextcloud/android/lib/resources/dashboard/DashboardGetWidgetItemsRemoteOperation.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Nextcloud Android client application
+ *
+ * @author Tobias Kaminsky
+ * Copyright (C) 2022 Tobias Kaminsky
+ * Copyright (C) 2022 Nextcloud GmbH
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program. If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.nextcloud.android.lib.resources.dashboard
+
+import com.google.gson.reflect.TypeToken
+import com.nextcloud.common.NextcloudClient
+import com.nextcloud.operations.GetMethod
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.ocs.ServerResponse
+import com.owncloud.android.lib.resources.OCSRemoteOperation
+import org.apache.commons.httpclient.HttpStatus
+
+class DashboardGetWidgetItemsRemoteOperation(val id: String) :
+    OCSRemoteOperation<HashMap<String, List<DashboardWidgetItem>>>() {
+    private val ENDPOINT = "/ocs/v2.php/apps/dashboard/api/v1/widget-items"
+
+    override fun run(client: NextcloudClient): RemoteOperationResult<HashMap<String, List<DashboardWidgetItem>>> {
+        lateinit var result: RemoteOperationResult<HashMap<String, List<DashboardWidgetItem>>>
+        lateinit var get: GetMethod
+
+        try {
+            get = GetMethod(client.baseUri.toString() + ENDPOINT + JSON_FORMAT, true)
+            get.setQueryString(mapOf(Pair("widgets[]", id), Pair("limit", "14")))
```

Comment:
Why this hardcoded limit?

Reply:
Designer said that we do not want to show a too long list, but rather then point to web.

Reply:
Shouldn't then this be a parameter for the operation, and then set when calling from the app? I think this is too restrictive for the library